components/browser/state/src/main/java/mozilla/components/browser/state/middleware/InProgressMediaMiddleware.kt

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.state.middleware
+
+import mozilla.components.browser.state.action.BrowserAction
+import mozilla.components.browser.state.action.EngineAction
+import mozilla.components.browser.state.action.MediaSessionAction
+import mozilla.components.browser.state.action.UpdateInProgressMediaTab
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.concept.engine.mediasession.MediaSession
+import mozilla.components.lib.state.Middleware
+import mozilla.components.lib.state.MiddlewareContext
+
+/**
+ * [Middleware] that updates [BrowserState.inProgressMediaTabId] everytime the user start playing media
+ * or navigates to another page in the last tab that was playing media.
+ */
+class InProgressMediaMiddleware : Middleware<BrowserState, BrowserAction> {
```

Comment:
Doing this in another Middleware adds quite a layer of indirection. If we really need this flag, wouldn't the media code handle this? 🤔

Reply:
Can we try and add move this to `feature-media` and handle it in a middleware there?